### PR TITLE
fftw: Re-enable OpenMP with non-GCC and musl

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

Clang now supports OpenMP, and musl has no problem with it either.

Following https://github.com/NixOS/nixpkgs/commit/3413562111a141cb7ffb687819dc9d63c52f11c7.

Related to #7023 (cc @acowley) and #34645.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

CC @spwhitt

### CODE REVIEW

**Comment by nh2** [comment]
I've tested by:

* building `pkgsMusl.fftw`
* building `pkgs.fftw` with `stdenv` replaced by `clangStdenv`

Still building:

* building `pkgsMusl.fftw` with `stdenv` replaced by `clangStdenv`

**Comment by nh2** [comment]
This needs a test on Darwin, can somebody please ofborg-build it for me?

File: pkgs/development/libraries/fftw/default.nix [hunk 1]
```diff
@@ -32,7 +38,7 @@ stdenv.mkDerivation {
     # all x86_64 have sse2
     # however, not all float sizes fit
     ++ optional (stdenv.isx86_64 && (precision == "single" || precision == "double") )  "--enable-sse2"
-    ++ optional (stdenv.cc.isGNU && !stdenv.hostPlatform.isMusl) "--enable-openmp"
+    ++ [ "--enable-openmp" ]
```
Comment (nh2) [inline pkgs/development/libraries/fftw/default.nix]: Keeping this here instead of moving it up to avoid mass-rebuilds on normal Linux.

**Review by acowley — APPROVED** [review]
- The change looks good to me
- It builds on darwin for me
- The build finds the `-fopenmp` flag

**Pull request merged**